### PR TITLE
Adding changelog.md to earthpy

### PR DESCRIPTION
**Pull request by lwasser**
Well I am not getting emails as our functions have changed. it's probably time to create this change log. @mbjoseph will you have a look at this please?

also @joemcglinchy @jlpalomino @mirob9363 @nkorinek  i'd love you to look at this as well to ensure i have captured all of our changes.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+**Updates to existing functions**
+
+Breaking Changes
+----------------
+
+* ``normalized_diff()`` function arguments have been flipped. Please update
+all of your old code accordingly. Changes include:
+    * arguments are now provided as normalized_diff(b1, b2)
+    * Math will be calculated (b1-b2) / (b1+b2)
+    * Example: ``ndvi = es.normalized_diff(b1=nir_band, b2=red_band)``
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
    * Example: `ndvi = es.normalized_diff(b1=nir_band, b2=red_band)`
```

File: CHANGES.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+* ``normalized_diff`` now returns unmasked array by default
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
* `normalized_diff` now returns unmasked array by default
```

File: CHANGES.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+* ``normalized_diff`` now returns masked array if there are nan values present
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
* `normalized_diff` now returns masked array if there are nan values present
```

File: CHANGES.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+* ``normalized_diff`` converts infinity values that result from division by zero to nan values
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
* `normalized_diff` converts infinity values that result from division by zero to nan values
```

File: CHANGES.rst [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+* ``stack_raster_tifs()`` is now deprecated and replaced with ``stack()``. You
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
* `stack()` replaces `stack_raster_tifs()`, which is now deprecated.
```

File: CHANGES.rst [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+should see a deprecation warning when calling it. The new ``stack()`` function
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
The new ``stack()`` function
```

File: CHANGES.rst [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+works similarly to ``stack_raster_tifs``. Inputs parameters are now a list of
+filepaths and an optional filepath parameter that when populated saves the
+stacked raster array as a geotiff file. The default return is a
+numpy ndarray.
+* The parameter order for ``bytescale`` has changed:
+   - PREVIOUS ORDER: data, cmin=None, cmax=None, high=255, low=0
+   - NEW ORDER: data, high=255, low=0, cmin=None, cmax=None
+
+**Plot Function Updates**
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
```

File: CHANGES.rst [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+All plot functions have moved to a new ``earthpy.plot`` module. To import plot
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
- All plot functions moved to a new ``earthpy.plot`` module. To import plot
```

Everything should be a bullet in a list (though we can have nested lists)

File: CHANGES.rst [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+functions first import the plot module: ``import earthpy.plot as ep``. Then you
+can call functions as follows:
+
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
```

File: CHANGES.rst [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+* ``ep.plot_rgb()``
```
Comment (mbjoseph) [inline CHANGES.rst]: These can be nested list elements

```suggestion
  * `ep.plot_rgb()`
```

File: CHANGES.rst [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+* ``ep.draw_legend()``
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
  * `ep.draw_legend()`
```

File: CHANGES.rst [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+* ``ep.hist()``
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
  * `ep.hist()`
```

File: CHANGES.rst [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+* ``ep.colorbar()``
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
  * `ep.colorbar()`
```

File: CHANGES.rst [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+* ``ep.plot_bands()``
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
  * `ep.plot_bands()`
```

File: CHANGES.rst [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+
+Enhancements
+------------
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
```

File: CHANGES.rst [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+* ``draw_legend()`` works now on different types of categorical raster plots.
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
* `draw_legend()` works now on different types of categorical raster plots.
```

File: CHANGES.rst [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+* ``colorbar()`` has also been fixed to work given updates to ``matplotlib``
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
* `colorbar()` has also been fixed to work given updates to Matplotlib
```

File: CHANGES.rst [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+* A new mask function has been released as ``mask_pixels()``.
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
* A new mask function has been released as `mask_pixels()`.
```

File: CHANGES.rst [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+
+**Infrastructure Updates**
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
```

File: CHANGES.rst [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+* **Enhanced testing:** We have updated our infrastructure to include ``pytest``
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
* We now have tests through pytest that run on Travis CI.
```

File: CHANGES.rst [hunk 21]
```diff
@@ -0,0 +1,66 @@
...
+run via TravisCI continuous integration for all testing.
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
```

File: CHANGES.rst [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+* **Docs:** We have implemented ``autodoc`` functionality through ReadTheDocs.io.
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
* API documentation is now generated automatically from docstrings with autodoc.
* Example code in documentation is tested using doctest
```

File: CHANGES.rst [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+This ensures that documentation is automatically build using function docstrings.
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
```

File: CHANGES.rst [hunk 24]
```diff
@@ -0,0 +1,66 @@
...
+
+Deprecations
+------------
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
```

File: CHANGES.rst [hunk 25]
```diff
@@ -0,0 +1,66 @@
...
+* ``stack_raster_tifs()`` has been deprecated and replaced with the ``stack()``
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
* `stack_raster_tifs()` has been deprecated and replaced with `stack()`.
```

File: CHANGES.rst [hunk 26]
```diff
@@ -0,0 +1,66 @@
...
+function. See details above.
```
Comment (mbjoseph) [inline CHANGES.rst]:
```suggestion
```

File: docs/change-log.rst [hunk 27]
```diff
@@ -0,0 +1 @@
+.. include:: ../CHANGES.rst
```
Comment (mbjoseph) [inline docs/change-log.rst]:
```suggestion
.. include:: ../CHANGELOG.md
```

File: docs/get-started.rst [hunk 28]
```diff
@@ -1,9 +1,9 @@
 Get Started With Earthpy
```
Comment (mbjoseph) [inline docs/get-started.rst]:
```suggestion
Get Started With EarthPy
```

File: docs/get-started.rst [hunk 29]
```diff
@@ -1,9 +1,9 @@
...
 ========================
 
-Earthpy is a python package devoted to working with spatial and remote sensing data.
-Earthpy Also contains an IO module that supports downloading data for the earth
-lab earth analytics courses.
+Earthpy is a python package devoted to working with spatial and remote sensing
```
Comment (mbjoseph) [inline docs/get-started.rst]:
```suggestion
EarthPy is a python package devoted to working with spatial and remote sensing
```

File: docs/get-started.rst [hunk 30]
```diff
@@ -1,9 +1,9 @@
...
+data. Earthpy also contains an IO module that supports downloading data for the
```
Comment (mbjoseph) [inline docs/get-started.rst]:
```suggestion
data. EarthPy also contains an IO module that supports downloading data for the
```

File: docs/get-started.rst [hunk 31]
```diff
@@ -1,9 +1,9 @@
...
+earth lab earth analytics courses.
```
Comment (mbjoseph) [inline docs/get-started.rst]:
```suggestion
Earth Lab earth analytics courses.
```

File: docs/get-started.rst [hunk 32]
```diff
@@ -1,9 +1,9 @@
...
 
 Earthpy Modules
```
Comment (mbjoseph) [inline docs/get-started.rst]:
```suggestion
EarthPy Modules
```

**Comment by lwasser** [comment]
hey @mbjoseph thank you for the review! 
i pushed a release today / last night. so i am thinking this list of changes could be associated with that. what do you think?? 0.6.2 on zenodo and pypi tag there is a tag. And yes in the future we can be more specific about changes related to releases!! this is great :)

**Comment by lwasser** [comment]
phew! 54 comments later - thank you @mbjoseph  i will merge this now. This closes: https://github.com/earthlab/earthpy/issues/195

**Pull request merged**